app/src/main/kotlin/compiler/ast/Expression.kt

```diff
@@ -47,6 +47,11 @@ sealed class Expression : AstNode {
         }
     }
 
+    data class ArrayGeneration(
+        val generatorCall: FunctionCall,
+        override val location: LocationRange? = null
+    ) : Expression()
+
```

Comment:
We had two options about "foreach in array" loop: syntax sugar or separate AST node. As far as I remember, we didn't settle on any option (so its up to the person who will implement that), so that person should be asked about that.